packages/host/app/services/matrix-service.ts

```diff
@@ -377,114 +379,149 @@ export default class MatrixService extends Service {
     body: string | undefined,
     attachedCards: CardDef[] = [],
     context?: OperatorModeContext,
+    clientGeneratedId?: string, // used to resend failed messages
   ): Promise<void> {
-    this.messagesToSend.set(roomId, undefined);
-    this.cardsToSend.set(roomId, undefined);
-    await this.setPendingMessage(roomId, body, attachedCards);
-
-    let html = markdownToHtml(body);
-    let functions = [];
-    let serializedAttachedCards: LooseSingleCardDocument[] = [];
-    let attachedOpenCards: CardDef[] = [];
-    let submode = context?.submode;
-    if (submode === 'interact') {
-      let mappings = await basicMappings(this.loaderService.loader);
-      // Open cards are attached automatically
-      // If they are not attached, the user is not allowing us to
-      // modify them
-      attachedOpenCards = attachedCards.filter((c) =>
-        (context?.openCardIds ?? []).includes(c.id),
+    try {
+      this.messagesToSend.set(roomId, undefined);
+      this.cardsToSend.set(roomId, undefined);
```

Comment:
i think the initialization of these two Maps is probably better performed inside of `this.setPendingMessage()`. Is there ever a time when you set a pending message that you would _not_ what to initialize this?

Reply:
Good idea! I can move these initializations to `setPendingMessage`